core/src/test/java/brooklyn/entity/rebind/RebindCatalogItemTest.java

```diff
@@ -192,6 +194,23 @@ public void testSameCatalogItemIdRemovalAndAdditionRebinds() throws Exception {
         rebindAndAssertCatalogsAreEqual();
     }
 
+    @Test
+    public void testRebindAfterItemDeprecated() {
+        String yaml =
+                "name: rebind-yaml-catalog-item-test\n" +
+                "brooklyn.catalog:\n" +
+                "  version: " + TEST_VERSION + "\n" +
+                "services:\n" +
+                "- type: io.camp.mock:AppServer";
+        BasicBrooklynCatalog catalog = (BasicBrooklynCatalog) origManagementContext.getCatalog();
+        CatalogItem<?, ?> catalogItem = catalog.addItem(yaml);
+        assertNotNull(catalogItem, "catalogItem");
+        catalogItem.setDeprecated(true);
```

Comment:
Does this need to also call `catalog.persist(catalogItem)`, after calling `setDeprecated`? Presumably it works just now because the add + setDeprecated happen in rapid succession, so the persister writes both changes in one go. If you put a sleep for 2 seconds in between then I'm betting it would fail.